src/v/cloud_storage_clients/s3_client.h

```diff
@@ -106,7 +107,14 @@ class request_creator {
       std::optional<char> delimiter = std::nullopt);
 
 private:
+    std::string make_host([[maybe_unused]] const bucket_name& name) const;
+
+    std::string make_target(
+      [[maybe_unused]] const bucket_name& name, const object_key& key) const;
```

Comment:
ss::sstring ?

Reply:
`Boost.Beast` expects an `std::string_view` in many places, such as `header.insert()` or `header.target()`, which is why this returns an `std::string` instead of an `ss::sstring`. 

I can refactor it to return an `ss::sstring` and pass it's data and length to `Boost.Beast` functions that expect a `string_view`, or it can remain an `std::string`- what do you think?

Reply:
let's leave it as std::string. thanks for the explantion.